ext/openssl/ossl.c

```diff
@@ -1150,9 +1150,29 @@ Init_openssl(void)
     /*
      * Version number of OpenSSL the ruby OpenSSL extension was built with
      * (base 16)
+     *
+     * OpenSSL 3: 0xMNN00PP0 (major minor 00 patch 0)
+     * OpenSSL 1: 0xMNNFFPPS (major minor fix patch status)
+     * LibreSSL : 0x20000000 (fixed value)
+     *
+     * See also the man page OPENSSL_VERSION_NUMBER(3).
      */
     rb_define_const(mOSSL, "OPENSSL_VERSION_NUMBER", INT2NUM(OPENSSL_VERSION_NUMBER));
 
+    /*
+     * Version number of LibreSSL the ruby OpenSSL extension was built with
+     * (base 16)
+     *
+     * LibreSSL: 0xMNNFFPPS (major minor fix patch status)
+     *
+     * See also the man page OPENSSL_VERSION_NUMBER(3).
+     */
+#if defined(LIBRESSL_VERSION_NUMBER)
+    rb_define_const(mOSSL, "LIBRESSL_VERSION_NUMBER", INT2NUM(LIBRESSL_VERSION_NUMBER));
+#else
+    rb_define_const(mOSSL, "LIBRESSL_VERSION_NUMBER", INT2NUM(0));
```

Comment:
I am setting the `OpenSSL::LIBRESSL_VERSION_NUMBER` here by `rb_define_const(mOSSL, "LIBRESSL_VERSION_NUMBER", INT2NUM(0));` if the macro `LIBRESSL_VERSION_NUMBER` is not defined, that is OpenSSL case. But I believe the right specification is the `OpenSSL::LIBRESSL_VERSION_NUMBER` is `nil` in the OpenSSL case. So, I will rebase this PR on the way.

Reply:
Rebased on the way.